src/main/java/edu/hm/hafner/analysis/IssueBuilder.java

```diff
@@ -28,6 +30,9 @@
  */
 @SuppressWarnings({"InstanceVariableMayNotBeInitialized", "JavaDocMethod", "PMD.TooManyFields"})
 public class IssueBuilder {
+
+    private static TreeStringBuilder treeStringBuilder = new TreeStringBuilder();
```

Comment:
While caching at class level is most effective I think that might be too risky at the moment (memory leak, synchronization). During parsing of an analysis report only a single instance of `IssueBuilder` is used so it should be sufficient to use a `TreeStringBuilder` per instance. So remove `static` for the moment, this is something we can tweak later on.